lib/vsc/mympirun/rm/sched.py

```diff
@@ -25,40 +25,60 @@
 """
 Main sched class
 """
-
-
+import pkgutil
 import os
+import time
 import random
 import re
-import time
-from vsc.utils.fancylogger import getLogger
-from vsc.mympirun.mpi.mpi import get_subclasses
+
 from vsc.utils.affinity import sched_getaffinity
-from vsc.utils.missing import nub
+from vsc.utils.fancylogger import getLogger
+from vsc.utils.missing import get_subclasses, nub
 
 
 def whatSched(requested):
     """Return the scheduler class"""
+
+    # import all modules in this dir: http://stackoverflow.com/a/16853487
+    for loader, modulename, _ in pkgutil.walk_packages([os.path.dirname(__file__)]):
+        loader.find_module(modulename).load_module(modulename)
+
     found_sched = get_subclasses(Sched)
+
+    # if there is no requested scheduler set or if the script is not in a job
+    # environment, default to local
+    if "PBS_JOBID" not in os.environ or requested is None:
```

Comment:
this is not correct. the `PBS_JOBID` is dependant on the rm subclass, you should loop over all classes and if none has envvar set, only then default to local